libs/parallelism/futures/include/hpx/futures/packaged_continuation.hpp

```diff
@@ -34,6 +34,7 @@ namespace hpx { namespace lcos { namespace detail {
     HPX_FORCEINLINE void transfer_result_impl(
         std::false_type, Source&& src, Destination& dest)
     {
+        hpx::intrusive_ptr<Destination> keep_alive(&dest);
```

Comment:
In particular the changes here in `packaged_continuation.hpp` look unnecessary to me. `try_catch_exception_ptr` doesn't do anything asynchronous on its own.

Reply:
It's not that `try_catch_exception_ptr` would do anything asynchronous. It's rather the `set_value`/`set_exception` calls that could trigger a chain of operations that eventually release the last reference to the shared state referred to by `dest`

Reply:
Hmm, ok. I suppose the question is then: why is it the responsibility of this particular function to keep the shared state alive?

Reply:
Good question. I went through the code base looking for all spots where continuations are invoked (directly or indirectly). We can certainly remove those for now and see if this breaks anything.